lib/puppet/daemon.rb

```diff
@@ -109,8 +112,8 @@ def set_signal_traps
     signals.update({:HUP => :restart, :USR1 => :reload, :USR2 => :reopen_logs }) unless Puppet.features.microsoft_windows?
     signals.each do |signal, method|
       Signal.trap(signal) do
-        Puppet.notice "Caught #{signal}; calling #{method}"
-        send(method)
+        Puppet.notice "Caught #{signal}; storing #{method}"
+        @signals << method
```

Comment:
Yep we came to the same conclusion in conversation lost in the default github view because of a subsequent push :)